docs/src/main/sphinx/installation/kubernetes.rst

```diff
@@ -0,0 +1,342 @@
+`Google Kubernetes Engine <https://cloud.google.com/kubernetes-engine>`_, and
+`Azure Kubernetes Service <https://azure.microsoft.com/en-us/services/kubernetes-service>`_.
+
+The fastest way to run Trino on Kubernetes is to use the
+`Trino Helm chart <https://github.com/trinodb/charts>`_. Helm is a package
+manager for Kubernetes applications that allows for simpler installation and
+versioning by templating Kubernetes configuration files. This allows you to
+prototype on your local or on-premise cluster and use the same deployment
+mechanism to deploy to the cloud to scale up.
+
+Prerequisites
+-------------
+
+* Have a running Kubernetes cluster with a version that falls within
+  `the Kubernetes support window <https://kubernetes.io/releases/>`_.
+
+  * For local deployments, we recommend using
+    `kind (Kubernetes in Docker) <https://kind.sigs.k8s.io>`_, which depends on
+    `Docker <https://www.docker.com>`_ to be installed as well. To run a
+    Kubernetes cluster in kind, run the command:
+
+    .. code-block:: text
+
+       kind create cluster --name trino
+
+    .. note::
+
+        The ``name`` parameter here is optional but is just used to showcase
+        how the namespace is applied in future commands. The cluster name
+        defaults to ``kind`` if no parameter is added. We recommend using
+        ``trino`` to make the application on this cluster obvious.
+
+* Install the `kubectl <https://kubernetes.io/docs/tasks/tools/#kubectl>`_
+  binary. Ensure you have a kubectl version that adheres to the
+  `Kubernetes version skew policy <https://kubernetes.io/releases/version-skew-policy/>`_.
+* Install the `helm <https://helm.sh>`_ binary. Ensure that you have a helm
+  version that adheres to the `Helm version skew policy <https://helm.sh/docs/topics/version_skew/>`_ that
+  supports the version of Kubernetes you are running.
+
+Running Trino using Helm
+------------------------
+
+Run the following commands from the system with ``helm`` and ``kubectl``
+installed and configured to connect to your running Kubernetes cluster:
+
+#. Validate kubectl is pointing to the correct cluster by running the command:
+
+   .. code-block:: text
+
+      kubectl cluster-info --context=kind-trino
+
+   .. note::
+
+      The ``context`` parameter here is optional and references the cluster
+      created in the example using kind above. ``kubectl`` will use the
+      `default context <https://kubernetes.io/docs/reference/kubectl/cheatsheet/#kubectl-context-and-configuration>`_
+      if this parameter isn't supplied. Notice the context is the name of the
+      cluster with the ``kind-`` prefix added.
+
+   You should see similar output with a random port assigned if you are
+   running on kind.
+
+   .. code-block:: text
+
+      Kubernetes control plane is running at https://127.0.0.1:59553
+      CoreDNS is running at https://127.0.0.1:59553/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
```

Comment:
Let's avoid this - we don't explain what the output means or how it's significant anyway.
How do I know from the output that I am using correct k8s cluster?
If we keep this then it needs a bit more explanation.

Also this output will differ depending on how you have k8s deployed - so another reason to have separate kind paragraph.

Reply:
I do agree having the specific output could be confusing so I'll remove that. I do think having a validation step where we say that they should be able to run a command that outputs the control plane url is fair. We link them to kubernetes documentation in the explanation above and I think it is out of scope to explain the details of this output. When they are getting started, seeing some output that is local or has a url containing the domain they're expecting should be simple enough.